services/src/main/java/org/fao/geonet/api/es/queryrewrite/FavouritesListESQueryRewriter.java

```diff
@@ -0,0 +1,299 @@
+                queryItems.remove(index);
+                queryItems.add(queryString);
+                fixsort((ObjectNode) root);
+                String result = mapper.writeValueAsString(root);
+                return result;
+            }
+            return jsonESQuery;
+        }
+        catch (Exception e) {
+            //don't modify
+            Log.debug(Geonet.SEARCH_ENGINE, "FavouritesListESQueryRewriter",e);
+            return jsonESQuery;
+        }
+    }
+
+    private void fixsort(ObjectNode root) {
+        if (!root.has("sort")) {
+            root.putArray("sort").add("_id");
+        }
+    }
+
+
+    /**
+     * constructs the
+     *
+     *  {
+     *    query:
+     *     {
+     *         query_string: "_id(uuid1) OR _id(uuid2) ..."
+     *     }
+     *  }
+     *
+     *  that will replace the
+     *
+     *           {
+     *                 "terms": {
+     *                   "favouritesList": [
+     *                        104
+     *                    ]
+     *                }
+     *            }
+     *
+     *   This checks security (i.e. does the user/session have read permission on the favourites list?)
+     */
+    ObjectNode replacement(int favouritesListId, ObjectMapper mapper, HttpSession httpSession, HttpServletRequest request) throws Exception {
+        Optional<FavouriteMetadataList> selectionList= userMetadataSelectionListRepo.findById(favouritesListId);
+        if (!selectionList.isPresent()) {
+            throw new Exception("could not find favouritesListId="+ favouritesListId);
+        }
+        //check security
+        boolean isAdmin = isAdmin(httpSession);
+        String sessionId = getSessionId(request);
+        User user =  getUser(httpSession);
+
+        if (!permittedRead(selectionList.get(),user,sessionId,isAdmin)) {
+            throw new Exception("not permitted to read");
+        }
+
+        List<String>  uuids = userMetadataSelectionRepo.queryByParent(favouritesListId);
+        List<String> statements = uuids.stream()
+            .map(x->"_id:("+x+")")
+            .collect(Collectors.toList());
+        String queryText = String.join(" OR ",statements);
+        ObjectNode query = mapper.createObjectNode();
+        ObjectNode queryString = mapper.createObjectNode();
+        queryString.put("query",queryText);
+        query.put("query_string",queryString);
+        return query;
+    }
+
+    /**
+     * Gets the user in the httpSession
+     */
+    User getUser(HttpSession httpSession) {
+        UserSession session = ApiUtils.getUserSession(httpSession);
+        String userId = session.getUserId(); // null = not logged on
+        User user = null;
+        if (userId != null) {
+            user = userRepository.findOne(userId);
+        }
+        return user;
+    }
+
+    /**
+     * returns true if the session represents an admin
+     */
+    boolean isAdmin(HttpSession httpSession) {
```

Comment:
Shouldn't it be `private`?
```suggestion
    private boolean isAdmin(HttpSession httpSession) {
```